src/ui/atoms/listItem/ListItem.tsx

```diff
@@ -3,57 +3,93 @@ import type { DeepReadonly } from 'superTypes'
 import { Typography } from '../typography/Typography'
 import './listItem.scss'
 import { isString } from 'utils'
+import classNames from 'classnames'
+import type { ListProps } from '../list/List'
 
-export type ListItemProps = {
+export type ListItemProps = Pick<ListProps, 'layout'> & {
   /**
    * The main title of the item.
    */
-  readonly title?: string
+  readonly title?: string | JSX.Element
   /**
    * Additional information about the item
    */
   readonly description?: string | JSX.Element
   /**
-   * An element that can provide visual information or allow interactions on the left side.
+   * An element that can provide visual information or allow interactions.
+   * If layout is in list mode, it will be displayed on the left side.
+   * If layout is in grid mode, it will be displayed on the top right corner.
```

Comment:
Maybe the description here should be shortened a bit because the firstElement (or lastElement) is not complex to understand, and it is a lot of text rendered in the Storybook properties table.

Reply:
I agree maybe you could just precise that its positioning is dependent on the `layout` property value ?